packages/svelte/src/reactivity/date.js

```diff
@@ -1,102 +1,132 @@
+/**
+ * @import { Source } from "#client"
+ */
+import { derived } from '../internal/client/index.js';
 import { source, set } from '../internal/client/reactivity/sources.js';
 import { get } from '../internal/client/runtime.js';
 
-/** @type {Array<keyof Date>} */
-const read = [
-	'getDate',
-	'getDay',
-	'getFullYear',
-	'getHours',
-	'getMilliseconds',
-	'getMinutes',
-	'getMonth',
-	'getSeconds',
-	'getTime',
-	'getTimezoneOffset',
-	'getUTCDate',
-	'getUTCDay',
-	'getUTCFullYear',
-	'getUTCHours',
-	'getUTCMilliseconds',
-	'getUTCMinutes',
-	'getUTCMonth',
-	'getUTCSeconds',
-	// @ts-expect-error this is deprecated
-	'getYear',
-	'toDateString',
-	'toISOString',
-	'toJSON',
-	'toLocaleDateString',
-	'toLocaleString',
-	'toLocaleTimeString',
-	'toString',
-	'toTimeString',
-	'toUTCString'
-];
-
-/** @type {Array<keyof Date>} */
-const write = [
-	'setDate',
-	'setFullYear',
-	'setHours',
-	'setMilliseconds',
-	'setMinutes',
-	'setMonth',
-	'setSeconds',
-	'setTime',
-	'setUTCDate',
-	'setUTCFullYear',
-	'setUTCHours',
-	'setUTCMilliseconds',
-	'setUTCMinutes',
-	'setUTCMonth',
-	'setUTCSeconds',
-	// @ts-expect-error this is deprecated
-	'setYear'
-];
+/**
+ * @typedef {Exclude<{[Key in keyof Date]: Date[Key] extends Function ? Key : never}[keyof Date], symbol>} DateMethodNames
+ */
 
 var inited = false;
-
 export class ReactiveDate extends Date {
+	/**
+	 * @type {Source<number>}
+	 */
 	#raw_time = source(super.getTime());
 
-	// We init as part of the first instance so that we can treeshake this class
+	/**
+	 * @type {Map<DateMethodNames, Source<unknown>>}
+	 */
+	#deriveds = new Map();
+
+	/**
+	 * @param {any[]} params
+	 */
+	constructor(...params) {
+		// @ts-ignore
+		super(...params);
+		this.#init();
+	}
+
 	#init() {
-		if (!inited) {
-			inited = true;
-			const proto = ReactiveDate.prototype;
-			const date_proto = Date.prototype;
+		if (inited) {
+			return;
+		}
+
+		inited = true;
+		var reactive_date_proto = ReactiveDate.prototype;
+		var date_proto = Date.prototype;
 
-			for (const method of read) {
+		/**
+		 * @type {Array<DateMethodNames>}
+		 */
+		var versioned_read_props = [
```

Comment:
Do these need to be separate, can't we just use the standard derived path for these too? I don't think the cost of avoiding a derived is worth it.

Reply:
They almost always change whenever the `time` changes, and if for example someone tries to use `toJSON` and `getTime` it won't create two separate signals for them. I thought creating the least amount of signals is something that should be considered if possible. Should I remove it?

Reply:
The aim is to create a minimal amount of signals, not create the least. It’s highly unlikely to use the Date object in a way that would really take advantage of this. It just adds more code size which is ultimately going to have fewer benefits.